### PR TITLE
Enhance `add-chain` tool to capture deploy config and monorep commit information 

### LINKED ISSUE #478: Capture genesis creation metadata into the registry
- `deploy-config.json`
- monorepo commit hash where genesis command was invoked
- command invocation and parameters for the genesis generation command

So that we can validate the genesis for standard chains.

We may as well store it for all chains, in case they want to become standard in the future.

### PR DESCRIPTION
**Pull request by geoknee**
Closes #478 

This PR targets the feature branch #489

### CODE REVIEW

File: add-chain/main.go [hunk 1]
```diff
@@ -227,3 +258,25 @@ func getGasPayingToken(l1rpcURl string, SystemConfigAddress superchain.Address)
...
+	}
+	return os.WriteFile(path.Join(targetDir, "deploy-config.json"), data, 0o777)
+}
+
+func writeGenesisValidationMetadata(chainId uint64, commit string, targetDir string) error {
+	vm := genesis.ValidationMetadata{
+		GenesisCreationCommit:  commit,
+		NodeVersion:            "18.12.1",
+		MonorepoBuildCommand:   "pnpm",
+		GenesisCreationCommand: "opnode1",
```
Comment (geoknee) [inline add-chain/main.go:275]: For now these will be hardcoded. If auto validation fails, these fields can be manually modified to try and get it to pass.

File: .env.example [hunk 2]
```diff
@@ -21,11 +21,18 @@ SCR_STANDARD_CHAIN_CANDIDATE=false
...
+# The following vars point to four input files required for adding a chain
 # Data will be scraped from these files in order to construct the required registry data
+# and for genesis validation purposes.
 SCR_DEPLOYMENTS_DIR=${SCR_MONOREPO_DIR}/packages/contracts-bedrock/deployments/getting-started
 SCR_ROLLUP_CONFIG=${SCR_MONOREPO_DIR}/op-node/rollup.json
 SCR_GENESIS=${SCR_MONOREPO_DIR}/op-node/genesis.json
+SCR_DEPLOY_CONFIG=${SCR_MONOREPO_DIR}/packages/contracts-bedrock/deploy-config/getting-started.json
+
+# This is the commit in the https://github.com/ethereum-optimism/optimism/ repo
+# at which the chain's genesis was created. This is necessary to have out validation checks
```
Comment (vdamle) [inline .env.example:33]:
```suggestion
# at which the chain's genesis was created. This is necessary to have our validation checks
```

File: add-chain/e2e_test.go [hunk 3]
```diff
@@ -10,9 +10,9 @@ import (
 )
 
 const (
-	addressDir             = "../superchain/extra/addresses/sepolia/"
-	configDir              = "../superchain/configs/sepolia/"
-	genesisSystemConfigDir = "../superchain/extra/genesis-system-configs/sepolia/"
+	addressDir         = "../superchain/extra/addresses/sepolia/"
+	configDir          = "../superchain/configs/sepolia/"
+	validtionInputsDir = "../validation/genesis/validation-inputs"
```
Comment (vdamle) [inline add-chain/e2e_test.go:15]:
```suggestion
	validationInputsDir = "../validation/genesis/validation-inputs"
```

(needs usages to be fixed as well)

File: add-chain/main.go [hunk 4]
```diff
@@ -227,3 +258,25 @@ func getGasPayingToken(l1rpcURl string, SystemConfigAddress superchain.Address)
 
 	return (*superchain.Address)(&result.Addr), nil
 }
+
+func copyDeployConfigFile(chainId uint64, sourcePath string, targetDir string) error {
+	data, err := os.ReadFile(sourcePath)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(path.Join(targetDir, "deploy-config.json"), data, 0o777)
```
Comment (vdamle) [inline add-chain/main.go:267]:
```suggestion
	return os.WriteFile(path.Join(targetDir, "deploy-config.json"), data, os.ModePerm)
```

File: add-chain/main.go (hunk 1 above)
Comment (vdamle) [reply add-chain/main.go:275]: Perhaps declare them as globals with a comment, so it'll catch attention easily?

File: add-chain/main.go [hunk 5]
```diff
@@ -227,3 +258,25 @@ func getGasPayingToken(l1rpcURl string, SystemConfigAddress superchain.Address)
...
+		GenesisCreationCommit:  commit,
+		NodeVersion:            "18.12.1",
+		MonorepoBuildCommand:   "pnpm",
+		GenesisCreationCommand: "opnode1",
+	}
+	data, err := toml.Marshal(vm)
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(path.Join(targetDir, "meta.toml"), data, 0o777)
```
Comment (vdamle) [inline add-chain/main.go:281]:
```suggestion
	return os.WriteFile(path.Join(targetDir, "meta.toml"), data, os.ModePerm)
```

**Review by vdamle — APPROVED** [review]
Approved with a few minor suggested tweaks.

**Pull request merged**